src/Application/Actions/DonorAccount/Create.php

```diff
@@ -45,21 +48,40 @@ protected function action(Request $request, Response $response, array $args): Re
         }
         \assert(is_array($requestBody));
 
-        $stripeCustomerId = $request->getAttribute(PersonManagementAuthMiddleware::PSP_ATTRIBUTE_NAME);
+        $stripeCustomerIdString = $request->getAttribute(PersonManagementAuthMiddleware::PSP_ATTRIBUTE_NAME);
+        \assert(is_string($stripeCustomerIdString));
 
-        $emailAddress = $requestBody['emailAddress'];
-        \assert(is_string($emailAddress) && is_string($stripeCustomerId));
+        $emailAddressString = $requestBody['emailAddress'] ?? throw new HttpBadRequestException($request, 'Expected emailAddress');
+        \assert(is_string($emailAddressString));
 
-        /** @var array{firstName: string, lastName: string} $donorName */
-        $donorName = $requestBody['donorName'];
+        /** @var array{firstName: string, lastName: string} $donorNameArray */
+        $donorNameArray = $requestBody['donorName'] ?? throw new HttpBadRequestException($request, 'Expected donorName');
+
+        try {
+            $emailAddress = EmailAddress::of($emailAddressString);
+            $donorName = DonorName::of($donorNameArray['firstName'], $donorNameArray['lastName']);
+            $stripeCustomerId = StripeCustomerId::of($stripeCustomerIdString);
```

Comment:
A nicer implementation would collect all exceptions and tell the client about them, instead of bailing out on the first one. We could look at valinor for that in future. But this is good enough for now I think. These should never happen unless there is a bug in frontend or someone is using different software to connect to matchbot.